taichi/program/state_flow_graph.cpp

```diff
@@ -155,14 +152,28 @@ StateToNodesMap::Range StateToNodesMap::operator[](const AsyncState &as) {
 }
 
 StateToNodesMap::Range StateToNodesMap::get_all_edges() {
-  maybe_sort();
+  if (empty()) {
+    // Note that we intentionally do NOT check |sorted_| if this is empty. This
+    // is fusion may call clear(), which sets |sorted_| to false. Later on,
+    // the cleared node will be disconnected, at which point this method gets
+    // invoked. Tricky..
+    // More things to note:
+    // * Following the above case, |sorted_| is also false for remove_node()
+    // during disconnection. However, because the range is empty, remove_node()
+    // is never really invoked...
+    // * Unlike here, operator[] unconditionally checks for |sorted_|.
+    return Range(data_.end(), data_.end());
```

Comment:
Would the logic here get simplified if we refactor `fuse` a bit?

Reply:
Ha, I feel like the logic here isn't that complicated. The bigger question here is to explain why we cannot asssert `sorted_` when this is empty..